apps/shinkai-desktop/src/pages/ais.tsx

```diff
@@ -514,14 +534,28 @@ const RemoveLLMProviderModal = ({
             </DialogClose>
             <Button
               className="min-w-[100px] flex-1"
-              isLoading={isPending}
+              isLoading={isPending || isLoadingLLMProviders}
               onClick={async () => {
                 if (!auth) return;
+                if (!isSuccessLLMProviders) {
+                  toast.error(t('llmProviders.errors.deleteAgent'), {
+                    description: t('llmProviders.errors.deleteAgent'),
+                  });
+                  return;
+                }
```

Comment:
I don't think we need this conditional, we already validated it in the listing component where we use `LLMProviderCard`